### ISSUE TITLE
[1.8]  [Using Forge build 1285] sending packets by using SimpleNetworkWrapper.sendToAll or sendToAllAround to multiple clients causes kick from server with errors in console

### ISSUE THREAD

**Issue by TechStack** [MinecraftForge/FML#588]
[FML version 8.0.14.1285]
Decoder Exception reported in console:
https://gist.github.com/TechStack/442357d160ee49ac2260

If the Packets are sent using sendToAllAround or SendToAll with only one player as the target it works fine (also works fine in SP for this reason).
But if multiple clients are online and are destinations of the packet then it results in the index out of range error.

I've tried back porting my project to 1.7 and I haven't encountered this problem in 1.7 yet so it appears to be 1.8 specific.
I can re-produce the error at will and i'm able to produce it using other 1.8 Mod also if it uses a call to send messages to multiple clients.

I've set break points in the FML code where it is causing the exception and I can see from watching the message object that it is in the processing of the modded packet but the payload appears to be in this "EmptyByteBuf" state. The EmptyByteBuf class is written to throw the indexoutofbounds error when the EmptyByteBuf.readByte method is called.

I haven't been successful in tracking down how the packet ended up being a EmptyByteBuf

MinecraftForge.net  thread discussing Issue (with Links to my project's source if you need an example place down a bulldozer from the creative menu and it will send the packets to everyone in the area):
http://www.minecraftforge.net/forum/index.php/topic,26817.0.html

Please advise what, if any additional info is needed.

**Comment by LexManos** [comment]
Actually this looks like a vanilla issue if I'm understanding the problem correctly.
Not resetting the buffer.
Would be nice if you could whip up a quick file to test out this issue instead of just linking and entire mod u.u

**Comment by TechStack** [comment]
Lex, Thanks for looking !

I'll check into creating a simpler test example with fewer moving parts either tonight or this weekend. 

I guess if it is a vanilla bug I could just loop thru the players online (or in the area) and send the packets one at a time. Not as elegant of a solution but its doable.


Cheers!

**Comment by LexManos** [comment]
If it is the vanilla bug I think it is I can easily fix, I just need something *simple* to reproduce it.

**Comment by TechStack** [comment]
Hi @LexManos I've created a new repo with a test case at:
https://github.com/TechStack/TestCaseFML

Build it & put the Jar on the dedicated server's Mods folder & run the dedicated server.

Launch 2 instances of the client and connect both to the server.
Ensure you are out of range of the spawn protection so the PlayerInteractEvent can fire correctly then click a block ( left or right) . The client will disconnect and have the IndexOutOfRange error in the console.

Hope this helps  I tried to keep it as simple as i could , its only 3 classes.
Thanks!
